### PR TITLE
Fix unittest not reporting on stdout

### PR DESCRIPTION
**Pull request by thetechnick**
### Change Type
Bug Fix

### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -635,20 +636,14 @@ func (Test) ValidateGitClean() {
...
-	if isCI {
-		// test output in json format
-		testCmd += " -json"
-	}
-	testCmd += " ./internal/... ./cmd/... ./apis/..."
-
-	if isCI {
-		testCmd = testCmd + " > " + locations.UnitTestExecReport()
-	}
+	testCmd := fmt.Sprintf("set -o pipefail; go test -coverprofile=%s -race -test.v=test2json", locations.UnitTestCoverageReport())
```
Comment (erdii) [inline magefile.go:639]: shouldn't the `set -o pipefail` be on the bash invocation further down?

File: magefile.go [hunk 2]
```diff
@@ -635,20 +636,14 @@ func (Test) ValidateGitClean() {
...
+	testCmd += " ./internal/... ./cmd/... ./apis/... "
+	testCmd += "| tee " + locations.UnitTestStdOut()
 
 	// cgo needed to enable race detector -race
-	must(sh.RunWithV(map[string]string{"CGO_ENABLED": "1"}, "bash", "-c", testCmd))
+	testErr := sh.RunWithV(map[string]string{"CGO_ENABLED": "1"}, "bash", "-c", testCmd)
+	must(sh.RunV("bash", "-c", "cat "+locations.UnitTestStdOut()+" | go tool test2json > "+locations.UnitTestExecReport()))
```
Comment (erdii) [inline magefile.go:645]:
```suggestion
	must(sh.RunV("bash", "-c", "set -o pipefail; cat "+locations.UnitTestStdOut()+" | go tool test2json > "+locations.UnitTestExecReport()))
```

File: magefile.go (hunk 1 above)
Comment (erdii) [reply magefile.go:639]:
```suggestion
	testCmd := fmt.Sprintf("set -o pipefail; go test -coverprofile=%s -race -test.v=test2json ./internal/... ./cmd/... ./apis/... | tee %s", locations.UnitTestCoverageReport(), locations.UnitTestStdOut())
```

File: magefile.go (hunk 2 above)
Comment (erdii) [inline magefile.go:645]:
```suggestion
	must(sh.RunV("bash", "-c", "set -o pipefail; cat "+locations.UnitTestStdOut()+" | go tool test2json > "+locations.UnitTestExecReport()))
```

File: docs/api-reference.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ apiVersion: package-operator.run/v1alpha1
 kind: ClusterObjectDeployment
 metadata:
   name: example
+  namespace: default
```
Comment (eqrx) [inline docs/api-reference.md:32]: This file is not part of the PR

**Review by eqrx — CHANGES_REQUESTED** [review]

**Review by erdii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**